spec/bugsnag_spec.rb

```diff
@@ -133,4 +133,153 @@ module Kernel
       Kernel.send(:remove_const, :REQUIRED)
     end
   end
+
+  describe "#leave_breadcrumb" do
+
+    let(:breadcrumbs) { Bugsnag.configuration.breadcrumbs }
+    let(:timestamp_regex) { /^\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:[\d\.]+Z$/ }
+
+    it "requires only a name argument" do
+      Bugsnag.leave_breadcrumb("TestName")
+      expect(breadcrumbs.to_a.size).to eq(1)
+      expect(breadcrumbs.first.to_h).to match({
+        :name => "TestName",
+        :type => Bugsnag::Breadcrumbs::MANUAL_BREADCRUMB_TYPE,
+        :metaData => {},
+        :timestamp => match(timestamp_regex)
+      })
+    end
+
+    it "accepts meta_data" do
+      Bugsnag.leave_breadcrumb("TestName", { :a => 1, :b => "2" })
+      expect(breadcrumbs.to_a.size).to eq(1)
+      expect(breadcrumbs.first.to_h).to match({
+        :name => "TestName",
+        :type => Bugsnag::Breadcrumbs::MANUAL_BREADCRUMB_TYPE,
+        :metaData => { :a => 1, :b => "2" },
+        :timestamp => match(timestamp_regex)
+      })
+    end
+
+    it "allows different message types" do
+      Bugsnag.leave_breadcrumb("TestName", {}, Bugsnag::Breadcrumbs::ERROR_BREADCRUMB_TYPE)
+      expect(breadcrumbs.to_a.size).to eq(1)
+      expect(breadcrumbs.first.to_h).to match({
+        :name => "TestName",
+        :type => Bugsnag::Breadcrumbs::ERROR_BREADCRUMB_TYPE,
+        :metaData => {},
+        :timestamp => match(timestamp_regex)
+      })
+    end
+
+    it "validates before leaving" do
+      Bugsnag.leave_breadcrumb(
+        "123123123123123123123123123123456456456456456456456456456456",
+        {
+          :a => 1,
+          :b => [1, 2, 3, 4],
+          :c => {
+            :test => true,
+            :test2 => false
+          }
+        },
+        "Not a real type"
+      )
+      expect(breadcrumbs.to_a.size).to eq(1)
+      expect(breadcrumbs.first.to_h).to match({
+        :name => "123123123123123123123123123123",
+        :type => Bugsnag::Breadcrumbs::MANUAL_BREADCRUMB_TYPE,
+        :metaData => {
+          :a => 1
+        },
+        :timestamp => match(timestamp_regex)
+      })
+    end
+
+    it "runs callbacks before leaving" do
+      Bugsnag.configuration.before_breadcrumb_callbacks << Proc.new { |breadcrumb|
```

Comment:
I normally see `do` + `end` for multi-line blocks instead of curly braces ("Avoid using {...} for multi-line blocks" at https://github.com/rubocop-hq/ruby-style-guide#single-line-blocks )